tests/api/test_upload_api.py

```diff
@@ -0,0 +1,76 @@
+import json
+import logging
+import os
+import re
+import shutil
+import tarfile
+import unittest
+import requests
+
+from insights.config import Settings
+logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.DEBUG)
+
+
+class UploadAPI(unittest.TestCase):
+    def setUp(self):
+        self.setting = Settings()
+        self.session = requests.session()
+        self.session.cert = self.setting.get_certs()
+        self.session.verify = False
+        self.base_url = self.setting.get('api', 'url')
+        self.archive_location = self.setting.get('upload_archive', 'archive_file_path')
+        self.search_file_path = self.setting.get('upload_archive', 'search_file_path')
+        self.bash_version = self.setting.get('upload_archive', 'bash_version')
+        self.machine_id_path = self.setting.get('upload_archive', 'machine_id_path')
+        self.new_bash_version = self.setting.get('upload_archive', 'new_bash_version')
+        self.un_archive_file_location = self.setting.get('upload_archive', 'un_archive_file_location')
+
+    def test_archive_upload(self):
+        """ Testing api upload endpoint """
+        self.get_system_id()
+        files = {'file': open(self.archive_location, "rb")}
+        self.upload = self.session.post(self.base_url + '/v1/uploads/' + self.system_id,
+                                        files=files)
+        assert self.upload.status_code == 201
+        logging.info("Upload done successfully")
+
+    def get_system_id(self):
```

Comment:
the methods for a specific purpose are to be written inside a module and not the test case.